app/src/main/java/org/mozilla/focus/utils/NewFeatureNotice.java

```diff
@@ -0,0 +1,68 @@
+package org.mozilla.focus.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+public class NewFeatureNotice {
+
+    private static final String PREF_KEY_BOOLEAN_FIRSTRUN_SHOWN = "firstrun_shown";
+    private static final String PREF_KEY_INT_FEATURE_UPGRADE_VERSION = "firstrun_upgrade_version";
+    private static final int MULTI_TAB_FROM_VERSION_1_0_TO_2_0 = 1;
+    private static final int FIREBASE_FROM_VERSION_2_0_TO_2_1 = 2;
+
+    private static NewFeatureNotice instance;
+
+    private final SharedPreferences preferences;
+
+    public synchronized static NewFeatureNotice getInstance(Context context) {
+        if (instance == null) {
+            instance = new NewFeatureNotice(context.getApplicationContext());
+        }
+        return instance;
+    }
+
+    private NewFeatureNotice(Context context) {
+        preferences = PreferenceManager.getDefaultSharedPreferences(context);
+    }
+
+    public boolean shouldShowMultiTabUpdate() {
+        return MULTI_TAB_FROM_VERSION_1_0_TO_2_0 == getLastShownFeatureVersion() + 1;
+    }
+
+    public void setMultiTabUpdateNoticeDidShow() {
+        preferences.edit()
+                .putBoolean(PREF_KEY_BOOLEAN_FIRSTRUN_SHOWN, true)
+                .putInt(PREF_KEY_INT_FEATURE_UPGRADE_VERSION, MULTI_TAB_FROM_VERSION_1_0_TO_2_0)
+                .apply();
+    }
+
+    public boolean shouldShowPrivacyPolicyUpdate() {
```

Comment:
I don't think this will ever be true.

Reply:
My bad. I forgot to flip it back after testing. Also having some minor modifications for the update path - 1.0.x to 2.1. Please help to review, thank you.